### PR TITLE
Remove write from access rights and add destroy right

### LINKED ISSUE #387: Add delete action to access rights resource
https://github.com/wintermeyer/omedis/issues/334

### PR DESCRIPTION
**Pull request by kagure-nyakio**
Fixes #387

### CODE REVIEW

File: lib/omedis/accounts/can_update_organisation.ex [hunk 1]
```diff
@@ -10,18 +10,29 @@ defmodule Omedis.Accounts.CanUpdateOrganisation do
...
+      tenant: organisation
+    )
+  end
+
+  def match?(actor, %{subject: %{data: organisation, action: %{type: :destroy}}}, _opts) do
+    Ash.exists?(
+      filter(
+        AccessRight,
+        resource_name == "Organisation" and
+          destroy && exists(group.group_memberships, user_id == ^actor.id)
```
Comment (almirsarajcic) [inline lib/omedis/accounts/can_update_organisation.ex:35]:
```elixir
def match?(actor, %{subject: %{data: organisation, action: %{type: action_type}}}, _opts)
      when action_type in [:update, :destroy] do
    Ash.exists?(
      filter(
        AccessRight,
        resource_name == "Organisation" and
          field(^action_type) and
          exists(group.group_memberships, user_id == ^actor.id)
      ),
      tenant: organisation
    )
  end
```
Screenshot 2024-11-26 at 16 54 55

File: lib/omedis/accounts/checks/can_access_resource.ex [hunk 2]
```diff
@@ -16,15 +16,57 @@ defmodule Omedis.Accounts.CanAccessResource do
   def match?(nil, _context, _options), do: false
   def match?(_actor, %{subject: %{tenant: nil}}, _options), do: false
 
-  def match?(actor, %{subject: %{tenant: organisation, resource: resource}} = context, _options) do
+  def match?(
```
Comment (almirsarajcic) [inline lib/omedis/accounts/checks/can_access_resource.ex:19]: Similar to above.

**Review by almirsarajcic — CHANGES_REQUESTED** [review]

File: lib/omedis/accounts/checks/can_access_resource.ex (hunk 2 above)
Comment (kagure-nyakio) [reply lib/omedis/accounts/checks/can_access_resource.ex:19]: I have already tried this. It doesn't work. Field(https://hexdocs.pm/ash/Ash.Resource.Info.html#field/2) returns an Ash.Resource.Info struct with the details of the attribute.

Comment (almirsarajcic) [reply lib/omedis/accounts/checks/can_access_resource.ex:19]: Ok. Find a solution that works.

**Comment by almirsarajcic** [comment]
@kagure-nyakio

File: lib/omedis/accounts/can_update_organisation.ex (hunk 1 above)
Comment (almirsarajcic) [reply lib/omedis/accounts/can_update_organisation.ex:35]: @kagure-nyakio, you missed tihs.

**Review by almirsarajcic — APPROVED** [review]

**Pull request merged**